addons/mrp/models/stock_orderpoint.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    show_bom = fields.Boolean('Show BoM column', compute='_compute_show_bom')
+    bom_id = fields.Many2one(
+        'mrp.bom', string='Bill of Materials',
+        domain="[('type', '=', 'normal'), '|', ('product_id', '=', product_id), '&', ('product_id', '=', False), ('product_tmpl_id', '=', product_tmpl_id)]")
+
+    def _compute_show_bom(self):
+        for orderpoint in self:
+            orderpoint.show_bom = any(ru.action == 'manufacture' for ru in orderpoint.route_id.rule_ids)
```

Comment:
maybe it would be more efficient to search_read the route_id of all 'action',  '=',  'manufacture' out of the loop? if you've got a report with 80 orderpoint it could be slow
not tested
``` python
route_contains_manufacture = defaultdict(lambda: False)
for res in self.env['stock.rule'].search_read([('action', '=', 'manufacture)], ['route_id'])
    route_contains_manufacture[res['route_id']] = True

for orderpoint in self:
    orderpoint.show_bom = route_contains_manufacture[orderpoint.route_id]
```